app/js/views/navigation-menu.jsx

```diff
@@ -1,19 +1,29 @@
 /* global React */
 
 export default class NavigationMenu extends React.Component {
+  constructor(props) {
+    this.foxbox = props.foxbox;
+  }
+
   shouldComponentUpdate() {
     // We never need to update this component as it is being recreated each time
     // the route changes.
     return false;
   }
 
+  handleOnClick(evt) {
+    this.foxbox.logout();
+
+    // Once logged out, we redirect to the login page.
+    location.hash = '#users/login';
```

Comment:
__nit:__ we can likely just set this hash directly on the link itself, in theory event handler is called first and only then default action is performed.